AccountEvmAddressMigration failure on testnet

### Description

AccountEvmAddressMigration fails on testnet

### Steps to reproduce

Run 1.58.7 migration

### Additional context

```bash
2022-05-09T13:25:15.120-0600 INFO main c.h.m.i.MirrorImporterApplication No active profile set, falling back to 1 default profile: "default"
2022-05-09T13:25:20.209-0600 INFO main o.f.c.i.c.DbMigrate Current version of schema "public": 1.58.6
2022-05-09T13:25:20.214-0600 INFO main o.f.c.i.c.DbMigrate Migrating schema "public" to version "1.58.7 - Populates evm_address for accounts with an ECDSA secp256k1 alias"
2022-05-09T13:25:20.897-0600 ERROR main o.f.c.i.c.DbMigrate Migration of schema "public" to version "1.58.7 - Populates evm_address for accounts with an ECDSA secp256k1 alias" failed! Changes successfully rolled back.
2022-05-09T13:25:20.901-0600 WARN main o.s.b.w.r.c.AnnotationConfigReactiveWebServerApplicationContext Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'flywayInitializer' defined in class path resource [org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration$FlywayConfiguration.class]: Invocation of init method failed; nested exception is org.flywaydb.core.internal.command.DbMigrate$FlywayMigrateException: Migration failed !
2022-05-09T13:25:20.967-0600 ERROR main o.s.b.SpringApplication Application run failed org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'flywayInitializer' defined in class path resource [org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration$FlywayConfiguration.class]: Invocation of init method failed; nested exception is org.flywaydb.core.internal.command.DbMigrate$FlywayMigrateException: Migration failed !
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1804)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322)
at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1154)
at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:908)
at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:64)
at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:740)
at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:415)
at org.springframework.boot.SpringApplication.run(SpringApplication.java:303)
at org.springframework.boot.SpringApplication.run(SpringApplication.java:1312)
at org.springframework.boot.SpringApplication.run(SpringApplication.java:1301)
at com.hedera.mirror.importer.MirrorImporterApplication.main(MirrorImporterApplication.java:33)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\\at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88)
Caused by: org.flywaydb.core.internal.command.DbMigrate$FlywayMigrateException: Migration failed !
at org.flywaydb.core.internal.command.DbMigrate.doMigrateGroup(DbMigrate.java:385)
at org.flywaydb.core.internal.command.DbMigrate.lambda$applyMigrations$1(DbMigrate.java:275)
at org.flywaydb.core.internal.jdbc.TransactionalExecutionTemplate.execute(TransactionalExecutionTemplate.java:55)
at org.flywaydb.core.internal.command.DbMigrate.applyMigrations(DbMigrate.java:274)
at org.flywaydb.core.internal.command.DbMigrate.migrateGroup(DbMigrate.java:247)
at org.flywaydb.core.internal.command.DbMigrate.lambda$migrateAll$0(DbMigrate.java:141)
at org.flywaydb.core.internal.database.postgresql.PostgreSQLAdvisoryLockTemplate.execute(PostgreSQLAdvisoryLockTemplate.java:69)
at org.flywaydb.core.internal.database.postgresql.PostgreSQLConnection.lock(PostgreSQLConnection.java:99)
at org.flywaydb.core.internal.schemahistory.JdbcTableSchemaHistory.lock(JdbcTableSchemaHistory.java:139)
at org.flywaydb.core.internal.command.DbMigrate.migrateAll(DbMigrate.java:141)
at org.flywaydb.core.internal.command.DbMigrate.migrate(DbMigrate.java:98)
at org.flywaydb.core.Flyway$1.execute(Flyway.java:172)
at org.flywaydb.core.Flyway$1.execute(Flyway.java:124)
at org.flywaydb.core.FlywayExecutor.execute(FlywayExecutor.java:205)
at org.flywaydb.core.Flyway.migrate(Flyway.java:124)
at org.springframework.boot.autoconfigure.flyway.FlywayMigrationInitializer.afterPropertiesSet(FlywayMigrationInitializer.java:66)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800)
 ...26 more
Caused by: org.flywaydb.core.api.FlywayException: Migration failed !
at org.flywaydb.core.internal.resolver.java.JavaMigrationExecutor.executeOnce(JavaMigrationExecutor.java:69)
at org.flywaydb.core.internal.resolver.java.JavaMigrationExecutor.lambda$execute$0(JavaMigrationExecutor.java:48)
at org.flywaydb.core.internal.database.DefaultExecutionStrategy.execute(DefaultExecutionStrategy.java:27)
at org.flywaydb.core.internal.resolver.java.JavaMigrationExecutor.execute(JavaMigrationExecutor.java:47)
at org.flywaydb.core.internal.command.DbMigrate.doMigrateGroup(DbMigrate.java:377)
   ... 43 more
Caused by: java.lang.IllegalArgumentException: Invalid point compression
at org.bouncycastle.math.ec.ECCurve$AbstractFp.decompressPoint(ECCurve.java:630)
at org.bouncycastle.math.ec.ECCurve.decodePoint(ECCurve.java:402)
at com.hedera.mirror.importer.util.Utility.decompressKey(Utility.java:95)
at com.hedera.mirror.importer.util.Utility.aliasToEvmAddress(Utility.java:76)
at com.hedera.mirror.importer.migration.AccountEvmAddressMigration.lambda$updateAlias$0(AccountEvmAddressMigration.java:57)
at org.springframework.jdbc.core.JdbcTemplate$RowCallbackHandlerResultSetExtractor.extractData(JdbcTemplate.java:1696)
at org.springframework.jdbc.core.JdbcTemplate$1.doInPreparedStatement(JdbcTemplate.java:723)
at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:651)
at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:713)
at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:738)
at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:768)
at org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate.query(NamedParameterJdbcTemplate.java:190)
at org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate.query(NamedParameterJdbcTemplate.java:202)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344)
at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:208)
at com.sun.proxy.$Proxy151.query(Unknown Source)
at com.hedera.mirror.importer.migration.AccountEvmAddressMigration.updateAlias(AccountEvmAddressMigration.java:54)
at com.hedera.mirror.importer.migration.AccountEvmAddressMigration.doMigrate(AccountEvmAddressMigration.java:43)
at com.hedera.mirror.importer.migration.MirrorBaseJavaMigration.migrate(MirrorBaseJavaMigration.java:59)
at org.flywaydb.core.internal.resolver.java.JavaMigrationExecutor.executeOnce(JavaMigrationExecutor.java:55)
     ... 47 more
```

### Hedera network

testnet

### Version

v0.56.1

### Operating system

_No response_

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/AccountEvmAddressMigration.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/AccountEvmAddressMigration.java
@@ -63,17 +63,22 @@ public class AccountEvmAddressMigration extends MirrorBaseJavaMigration {
     }
 
     @Override
+    public Integer getChecksum() {
+        return 1; // Change this if this migration should be rerun
+    }
+
+    @Override
     public String getDescription() {
         return "Populates evm_address for accounts with an ECDSA secp256k1 alias";
     }
 
     @Override
-    public MigrationVersion getVersion() {
-        return null;
+    public MigrationVersion getMinimumVersion() {
+        return MigrationVersion.fromVersion("1.58.6");
     }
 
     @Override
-    public Integer getChecksum() {
-        return 1; // Change this if this migration should be rerun
+    public MigrationVersion getVersion() {
+        return null;
     }
 }

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/AccountEvmAddressMigration.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/AccountEvmAddressMigration.java
@@ -69,6 +69,11 @@ public class AccountEvmAddressMigration extends MirrorBaseJavaMigration {
 
     @Override
     public MigrationVersion getVersion() {
-        return MigrationVersion.fromVersion("1.58.7");
+        return null;
+    }
+
+    @Override
+    public Integer getChecksum() {
+        return 1; // Change this if this migration should be rerun
     }
 }

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/util/Utility.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/util/Utility.java
@@ -23,14 +23,12 @@ package com.hedera.mirror.importer.util;
 import com.google.common.collect.Iterables;
 import com.google.protobuf.ByteString;
 import com.google.protobuf.GeneratedMessageV3;
-import com.google.protobuf.InvalidProtocolBufferException;
 import com.google.protobuf.TextFormat;
 import com.hederahashgraph.api.proto.java.AccountID;
 import com.hederahashgraph.api.proto.java.ContractLoginfo;
 import com.hederahashgraph.api.proto.java.Key;
 import com.hederahashgraph.api.proto.java.Timestamp;
 import com.hederahashgraph.api.proto.java.TransactionID;
-import java.math.BigInteger;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.StandardOpenOption;
@@ -38,7 +36,7 @@ import java.time.Instant;
 import java.util.Arrays;
 import lombok.experimental.UtilityClass;
 import lombok.extern.log4j.Log4j2;
-import org.bouncycastle.asn1.x9.X9IntegerConverter;
+import org.apache.commons.codec.binary.Hex;
 import org.bouncycastle.jcajce.provider.digest.Keccak;
 import org.bouncycastle.math.ec.ECCurve;
 import org.bouncycastle.math.ec.ECPoint;
@@ -53,7 +51,6 @@ public class Utility {
 
     public static final Instant MAX_INSTANT_LONG = Instant.ofEpochSecond(0, Long.MAX_VALUE);
     private static final ECCurve SECP256K1_CURVE = new SecP256K1Curve();
-    private static final X9IntegerConverter X9_CONVERTER = new X9IntegerConverter();
 
     /**
      * Converts an ECDSA secp256k1 alias to a 20 byte EVM address by taking the keccak hash of it. Logic copied from
@@ -72,27 +69,19 @@ public class Utility {
 
             if (key.getKeyCase() == Key.KeyCase.ECDSA_SECP256K1) {
                 var rawCompressedKey = DomainUtils.toBytes(key.getECDSASecp256K1());
-                BigInteger x = new BigInteger(rawCompressedKey, 1, 32);
-                ECPoint ecPoint = decompressKey(x, (rawCompressedKey[0] & 0x1) == 0x1);
+                ECPoint ecPoint = SECP256K1_CURVE.decodePoint(rawCompressedKey);
                 byte[] uncompressedKeyDer = ecPoint.getEncoded(false);
                 byte[] uncompressedKeyRaw = new byte[64];
                 System.arraycopy(uncompressedKeyDer, 1, uncompressedKeyRaw, 0, 64);
                 byte[] hashedKey = new Keccak.Digest256().digest(uncompressedKeyRaw);
-
                 return Arrays.copyOfRange(hashedKey, 12, 32);
             }
 
             return null;
-        } catch (InvalidProtocolBufferException e) {
-            throw new ParserException(e);
+        } catch (Exception e) {
+            var aliasHex = Hex.encodeHexString(alias);
+            throw new ParserException("Unable to decode alias to EVM address: " + aliasHex, e);
         }
-    }
-
-    // Decompress a compressed public key (x coordinate and low-bit of y-coordinate).
-    private static ECPoint decompressKey(BigInteger x, boolean yBit) {
-        final byte[] compEnc = X9_CONVERTER.integerToBytes(x, X9_CONVERTER.getByteLength(SECP256K1_CURVE) + 1);
-        compEnc[0] = (byte) (yBit ? 0x03 : 0x02);
-        return SECP256K1_CURVE.decodePoint(compEnc);
     }
 
     /**

--- a/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/AccountEvmAddressMigrationTest.java
+++ b/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/AccountEvmAddressMigrationTest.java
@@ -20,8 +20,8 @@ package com.hedera.mirror.importer.migration;
  * ‍
  */
 
-import static com.hedera.mirror.importer.util.UtilityTest.EVM_ADDRESS;
 import static com.hedera.mirror.importer.util.UtilityTest.ALIAS_ECDSA_SECP256K1;
+import static com.hedera.mirror.importer.util.UtilityTest.EVM_ADDRESS;
 import static org.assertj.core.api.Assertions.assertThat;
 
 import lombok.RequiredArgsConstructor;
@@ -30,15 +30,11 @@ import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.jdbc.core.JdbcOperations;
-import org.springframework.test.context.TestPropertySource;
 
-import com.hedera.mirror.importer.EnabledIfV1;
 import com.hedera.mirror.importer.IntegrationTest;
 
-@EnabledIfV1
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 @Tag("migration")
-@TestPropertySource(properties = "spring.flyway.target=1.58.6")
 class AccountEvmAddressMigrationTest extends IntegrationTest {
 
     private final JdbcOperations jdbcOperations;

--- a/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/util/UtilityTest.java
+++ b/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/util/UtilityTest.java
@@ -49,8 +49,12 @@ public class UtilityTest {
     @Test
     void aliasToEvmAddress() {
         byte[] aliasEd25519 = Key.newBuilder().setEd25519(ByteString.copyFromUtf8("ab")).build().toByteArray();
-        byte[] invalidBytes = new byte[]{'a', 'b'};
+        byte[] aliasEcdsa2 = Hex.decode("3a2102ff806fecbd31b4c377293cba8d2b78725965a4990e0ff1b1b29a1d2c61402310");
+        byte[] evmAddress2 = Hex.decode("efa0d905af20199aa03aca71cfa5f7647f29f439");
+        byte[] invalidBytes = new byte[] {'a', 'b'};
+
         assertThat(Utility.aliasToEvmAddress(ALIAS_ECDSA_SECP256K1)).isEqualTo(EVM_ADDRESS);
+        assertThat(Utility.aliasToEvmAddress(aliasEcdsa2)).isEqualTo(evmAddress2);
         assertThat(Utility.aliasToEvmAddress(aliasEd25519)).isNull();
         assertThat(Utility.aliasToEvmAddress(null)).isNull();
         assertThat(Utility.aliasToEvmAddress(new byte[] {})).isNull();

